### PR TITLE
Fixes whitespace issue with token, causing them to be invalid

### PR DESCRIPTION
**Pull request by namratachaudhary**

### CODE REVIEW

File: pkg/controller/oneagent/util.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func getToken(secret *corev1.Secret, key string) (string, error) {
 		return "", err
 	}
 
-	return string(value), nil
+	return strings.TrimSpace(value), nil
```
Comment (esukram) [inline pkg/controller/oneagent/util.go]: One could use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L4992
`Secret.stringData` to have the `string` representation.

Comment (baichinger) [reply pkg/controller/oneagent/util.go]: `StringData` is write-only and only used for creating secrets. Don't see viable alternatives to `strings.TrimSpace` or `bytes.TrimSpace`.

Comment (esukram) [reply pkg/controller/oneagent/util.go]: Wasn't aware this fact.

File: pkg/controller/oneagent/util.go [hunk 2]
```diff
@@ -127,7 +127,7 @@ func getToken(secret *corev1.Secret, key string) (string, error) {
 		return "", err
 	}
 
-	return strings.TrimSpace(value), nil
+	return strings.TrimSpace(string(value)), nil
```
Comment (esukram) [inline pkg/controller/oneagent/util.go]: Works.

File: pkg/controller/oneagent/util.go (hunk 1 above)
Comment (namratachaudhary) [reply pkg/controller/oneagent/util.go]: My IDE failed to give this error on compilation. Fixed it, should be able to catch it next time.

**Comment by namratachaudhary** [comment]
@esukram @lrgar Added tests, there's a comment [here](https://github.com/Dynatrace/dynatrace-oneagent-operator/pull/77/files#diff-d95cdfb368554bd6faec472e6089b802R72) where I had doubt, please check.

**Comment by esukram** [comment]
> @esukram @lrgar Added tests, there's a comment [here](https://github.com/Dynatrace/dynatrace-oneagent-operator/pull/77/files#diff-d95cdfb368554bd6faec472e6089b802R72) where I had doubt, please check.

Then we should rethink returning an error if the token is empty. This won't work.

cc/ @baichinger

File: pkg/controller/oneagent/util_test.go [hunk 3]
```diff
@@ -62,6 +62,43 @@ func TestOneAgent_Validate(t *testing.T) {
...
+		data["test_token_2"] = []byte("\t\n   dynatrace_test_token_2")
+		secret := corev1.Secret{Data: data}
+		token, err := getToken(&secret, "test_token")
+		token2, err := getToken(&secret, "test_token_2")
+
+		assert.NoError(t, err)
+		assert.Equal(t, token, "dynatrace_test_token")
+		assert.Equal(t, token2, "dynatrace_test_token_2")
+	}
+}
```
Comment (lrgar) [inline pkg/controller/oneagent/util_test.go]: Nitpicking: new line between functions.

**Pull request merged**